project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -139,6 +145,45 @@ public String getAllQuestionsJSON(String key) {
     return new Gson().toJson("");
   }
 
+/** 
+ * this function takes the assignment Key and answer group key as parameters
+ * returns list of blobKeys to submissions that fit in specified answer group for specific question
+ */ 
+  public List blobkeysFromAnswerGroup(String assignKey, String groupKey) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // query answer entities filtered only in group from parameter
+    Filter groupFilter = new FilterPredicate("groupKey", FilterOperator.EQUAL, groupKey);
+    Query answersQuery = new Query("Answer").setFilter(groupFilter);
+    PreparedQuery answerEntities = datastore.prepare(answersQuery);
+
+    // generate list of submissionKeys for the group
+    List<String> submissionKeys = new ArrayList<>();
+    for (Entity entity : answerEntities.asIterable()) {
+        String submissionKey = (String) entity.getProperty("submissionKey");
+        submissionKeys.add(submissionKey);
+    }  
+
+    // query submissions filtered to the assignment
+    Filter assignmentKeyFilter = new FilterPredicate("assignmentKey", FilterOperator.EQUAL, assignKey);
+    Query submissionsQuery = new Query("Submission").setFilter(assignmentKeyFilter);
+    PreparedQuery submissionEntities = datastore.prepare(submissionsQuery);
+    // assignmentKeyFilter is unneeded. I'm unclear if adding this filter will speedup or slowdown the program
```

Comment:
I agree its not needed and can be removed safely

Reply:
ok